Sources/PaymentButtons/ImageAsset.swift

```diff
@@ -28,7 +47,15 @@ enum ImageAsset {
         case .blue:
             imageAssetString += "blue"
         }
-
-        return UIImage(named: imageAssetString, in: PaymentButton.bundle, compatibleWith: nil)
+        
+        return imageAssetString
+    }
+    
+    private let button: PaymentButton
+    
+    // MARK: - Initializer
+        
+    init(button: PaymentButton) {
+        self.button = button
     }
```

Comment:
```
init(fundingSource: FundingSource) {
    self.fundingSource = fundingSource
}
```
nit: take or leave we could pass the funding source and omit the button reference here.

Reply:
We also reference the button.color ([code](https://github.com/paypal/paypal-ios/blob/73e3e0c5f80f0019254e1c3ac9ba3b755d7c4729/Sources/PaymentButtons/ImageAsset.swift#L40)) & button.size ([code](https://github.com/paypal/paypal-ios/blob/73e3e0c5f80f0019254e1c3ac9ba3b755d7c4729/Sources/PaymentButtons/ImageAsset.swift#L29)), in addition to the fundingSource.

The current setup doesn't feel like the best design to me. An extension on PaymentButton probably makes more sense than a separate class, or breaking up PaymentButton to separate responsibilities out more.

Reply:
I went with an extension (see commit 957ffdcb7f7bab36bad96ecf1491875cdf60aa40) for now. Let me know what you think!

Reply:
👏 🚢